rust-webvr/src/api/googlevr/service.rs

```diff
@@ -171,45 +145,37 @@ impl GoogleVRService {
         return !self.ctx.is_null();
     }
 
-    fn clone_displays(&self) -> Vec<VRDisplayPtr> {
-        self.displays.iter().map(|d| d.clone() as VRDisplayPtr).collect()
-    }
+    fn init_display(&mut self) -> Result<&GoogleVRDisplayPtr, String> {
+        self.initialize()?;
 
-    fn clone_gamepads(&self) -> Vec<VRGamepadPtr> {
-        self.gamepads.iter().map(|d| d.clone() as VRGamepadPtr).collect()
+        if let Some(ref d) = self.display {
+            Ok(d)
+        } else {
+            self.display = unsafe { Some(GoogleVRDisplay::new(self, self.ctx, self.controller_ctx)) };
+            Ok(self.display.as_ref().unwrap())
+        }
```

Comment:
Can we use `Option::get_or_insert_with`?

Reply:
Tried that, we borrow `self` here so we can't